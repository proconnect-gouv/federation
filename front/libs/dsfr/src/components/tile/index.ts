/* istanbul ignore file */

// declarative file
export * from './tile.component';
