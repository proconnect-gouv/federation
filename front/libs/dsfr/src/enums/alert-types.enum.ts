/* istanbul ignore file */
export enum AlertTypes {
  ERROR = 'error',
  INFO = 'info',
  SUCCESS = 'success',
  WARNING = 'warning',
}
