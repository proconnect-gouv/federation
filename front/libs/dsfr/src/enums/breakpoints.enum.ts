/* istanbul ignore file */
export enum Breakpoints {
  XS = 576,
  SM = 768,
  MD = 992,
  LG = 1248,
}
