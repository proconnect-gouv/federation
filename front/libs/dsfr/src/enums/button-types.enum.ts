/* istanbul ignore file */
export enum ButtonTypes {
  SUBMIT = 'submit',
  BUTTON = 'button',
  RESET = 'reset',
}
