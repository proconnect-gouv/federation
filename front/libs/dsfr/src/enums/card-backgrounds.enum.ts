export enum CardBackgrounds {
  GREY = 'grey',
  NO_BACKGROUND = 'no-background',
  NO_BORDER = 'no-border',
  SHADOW = 'shadow',
}
