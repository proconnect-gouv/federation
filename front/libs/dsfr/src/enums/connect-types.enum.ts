export enum ConnectTypes {
  FRANCE_CONNECT = 'FranceConnect',
  PRO_CONNECT = 'ProConnect',
}
