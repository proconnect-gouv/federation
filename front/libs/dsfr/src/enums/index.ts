/* istanbul ignore file */

// declarative file
export * from './alert-types.enum';
export * from './button-types.enum';
export * from './icon-placement.enum';
export * from './pagination.enum';
export * from './priorities.enum';
export * from './sizes.enum';
