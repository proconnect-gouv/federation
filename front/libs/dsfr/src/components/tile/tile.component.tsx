import classnames from 'classnames';
import type { PropsWithChildren } from 'react';
import React from 'react';
import { Link } from 'react-router-dom';

import { HeadingTag } from '@fc/common';

import { Sizes } from '../../enums';

interface TileComponentProps extends PropsWithChildren {
  isHorizontal?: boolean;
  useDownload?: boolean | undefined;
  description?: string;
  dataTestId?: string | undefined;
  detail?: string;
  Heading?: HeadingTag;
  link: string;
  size?: Sizes;
  title: string;
}

// @TODO
// - force display mode (.fr-tile--vertical@md | fr-tile--horizonral@md)
// - use download button instead of line (.fr-enlarge-button)
// @SEE https://www.systeme-de-design.gouv.fr/composants-et-modeles/composants/tuile
export const TileComponent = React.memo(
  ({
    Heading = HeadingTag.H3,
    children: svgArtwork,
    dataTestId = 'TileComponent',
    description = undefined,
    detail = undefined,
    isHorizontal = false,
    link,
    size = Sizes.MEDIUM,
    title,
    useDownload = false,
  }: TileComponentProps) => (
    <div
      className={classnames(`fr-tile fr-tile--${size} fr-enlarge-link`, {
        // @NOTE CSS class
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-tile--download': useDownload,
        // @NOTE CSS class
        // eslint-disable-next-line @typescript-eslint/naming-convention
        'fr-tile--horizontal': isHorizontal,
      })}
      data-testid={dataTestId}>
      <div className="fr-tile__body">
        <div className="fr-tile__content">
          <Heading className="fr-tile__title" data-testid={`${dataTestId}-title`}>
            <Link download={useDownload} to={link}>
              {title}
            </Link>
          </Heading>
          {description && <p className="fr-tile__desc">{description}</p>}
          {detail && <p className="fr-tile__detail">{detail}</p>}
          {/* @TODO add badges / tags */}
        </div>
      </div>
      {svgArtwork && (
        <div className="fr-tile__header">
          <div className="fr-tile__pictogram">{svgArtwork}</div>
        </div>
      )}
    </div>
  ),
);

TileComponent.displayName = 'TileComponent';
