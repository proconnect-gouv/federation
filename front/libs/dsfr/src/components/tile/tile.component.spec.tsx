import { render } from '@testing-library/react';

import { HeadingTag } from '@fc/common';

import { Sizes } from '../../enums';
import { TileComponent } from './tile.component';

describe('TileComponent', () => {
  it('should match the snapshot, with default values', () => {
    // When
    const { container } = render(
      <TileComponent link="any-link-mock" title="Tile title mock">
        any description text treat as children
      </TileComponent>,
    );

    // Then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot, with optionnal values', () => {
    // When
    const { container } = render(
      <TileComponent
        isHorizontal
        useDownload
        dataTestId="any-data-test-id-mock"
        description="any-description-mock"
        detail="any-detail-mock"
        Heading={HeadingTag.H1}
        link="any-link-mock"
        size={Sizes.LARGE}
        title="Tile title mock">
        any description text treat as children
      </TileComponent>,
    );

    // Then
    expect(container).toMatchSnapshot();
  });
});
