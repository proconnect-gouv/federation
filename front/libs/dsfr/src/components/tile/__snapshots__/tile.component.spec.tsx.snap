// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileComponent should match the snapshot, with default values 1`] = `
<div>
  <div
    class="fr-tile fr-tile--md fr-enlarge-link"
    data-testid="TileComponent"
  >
    <div
      class="fr-tile__body"
    >
      <div
        class="fr-tile__content"
      >
        <h3
          class="fr-tile__title"
          data-testid="TileComponent-title"
        >
          <span
            data-mockid="Link"
          >
            Tile title mock
          </span>
        </h3>
      </div>
    </div>
    <div
      class="fr-tile__header"
    >
      <div
        class="fr-tile__pictogram"
      >
        any description text treat as children
      </div>
    </div>
  </div>
</div>
`;

exports[`TileComponent should match the snapshot, with optionnal values 1`] = `
<div>
  <div
    class="fr-tile fr-tile--lg fr-enlarge-link fr-tile--download fr-tile--horizontal"
    data-testid="any-data-test-id-mock"
  >
    <div
      class="fr-tile__body"
    >
      <div
        class="fr-tile__content"
      >
        <h1
          class="fr-tile__title"
          data-testid="any-data-test-id-mock-title"
        >
          <span
            data-mockid="Link"
          >
            Tile title mock
          </span>
        </h1>
        <p
          class="fr-tile__desc"
        >
          any-description-mock
        </p>
        <p
          class="fr-tile__detail"
        >
          any-detail-mock
        </p>
      </div>
    </div>
    <div
      class="fr-tile__header"
    >
      <div
        class="fr-tile__pictogram"
      >
        any description text treat as children
      </div>
    </div>
  </div>
</div>
`;
